app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -437,5 +439,21 @@ def concern_has_file_sets?
     def uploaded_files
       UploadedFile.find(params.fetch(:uploaded_files, []))
     end
+
+    def available_admin_sets
+      admin_set_results = Hyrax::AdminSetService.new(self).search_results(:deposit)
```

Comment:
Do we need a branch on this initial AdminSet/AdministrativeSet query to support different query backends?

Reply:
i don't think so. i think this is all Solr based, and using blacklight infrastructure.

unless/until Fedora supports a fast lookup for all admin sets filtered on deposit access, i think we probably want it to stay that way.